### PR TITLE
[k8s-otel][docs]Custom configuration and upgrade sections

### PR DESCRIPTION
**Pull request by eedugon**
Added the following sub-sections to Kubernetes onboarding documentation 👍 

- Upgrades
- Customizing Configuration, including ClusterName and other relevant configuration settings that could be tweaked.
- It also includes a small fix about the TLS self-signed certificate and cert-manager by removing the `publicly trusted` statement, as even if cert-manager is used the certificate will still be `self-signed` (although automatically renewed).

I'll be requesting in this PR first a review by docs team and then by product team (@rogercoll / @gizas / @mlunadia ).

Closes https://github.com/elastic/opentelemetry/issues/35

### CODE REVIEW

File: docs/kubernetes/operator/README.md [hunk 1]
```diff
@@ -181,18 +183,63 @@ For detailed instructions and examples on how to instrument applications in Kube
 
 For troubleshooting details and verification steps, refer to [Troubleshooting auto-instrumentation](/docs/kubernetes/operator/troubleshoot-auto-instrumentation.md).
 
+<a name="operator-upgrade"></a>
+
+## Upgrades
```
Comment (mdbirnstiehl) [inline docs/kubernetes/operator/README.md:188]:
```suggestion
## Upgrade
```

File: docs/kubernetes/operator/README.md [hunk 2]
```diff
@@ -15,6 +15,8 @@ This guide describes how to:
 - [Manual deployment of all components](#manual-deployment-of-all-components)
 - [Installation verification](#installation-verification)
 - [Instrumenting applications](#instrumenting-applications)
+- [Upgrades](#operator-upgrade)
```
Comment (mdbirnstiehl) [inline docs/kubernetes/operator/README.md:18]:
```suggestion
- [Upgrade](#operator-upgrade)
```

File: docs/kubernetes/operator/README.md [hunk 3]
```diff
@@ -181,18 +183,63 @@ For detailed instructions and examples on how to instrument applications in Kube
 
 For troubleshooting details and verification steps, refer to [Troubleshooting auto-instrumentation](/docs/kubernetes/operator/troubleshoot-auto-instrumentation.md).
 
+<a name="operator-upgrade"></a>
+
+## Upgrades
+
+> [!NOTE]
+> Before upgrading or updating the release configuration refer to [compatibility matrix](#compatibility-matrix) for the list of supported versions and [customizing configuration](#custom-configuration) for a list of supported configurable parameters.
```
Comment (mdbirnstiehl) [inline docs/kubernetes/operator/README.md:191]:
```suggestion
> Before upgrading or updating the release configuration refer to [compatibility matrix](#compatibility-matrix) for a list of supported versions and [customizing configuration](#custom-configuration) for a list of supported configurable parameters.
```
For consistent structure with the clause before (for the list/for a list).

File: docs/kubernetes/operator/README.md [hunk 4]
```diff
@@ -181,18 +183,63 @@ For detailed instructions and examples on how to instrument applications in Kube
...
+
+```bash
+helm repo update open-telemetry # update information of available charts locally
+helm search repo open-telemetry/opentelemetry-kube-stack --versions # list available versions of the chart
+
+helm upgrade --namespace opentelemetry-operator-system opentelemetry-kube-stack open-telemetry/opentelemetry-kube-stack \
+ --values <updated_values_file> --version <updated_version>
+```
+
+If [cert-manager integration](#cert-manager) is disabled, helm will generate a new self-signed TLS certificate with every update, even if there are no actual changes to apply.
```
Comment (mdbirnstiehl) [inline docs/kubernetes/operator/README.md:203]:
```suggestion
If [cert-manager integration](#cert-manager) is disabled, Helm generates a new self-signed TLS certificate with every update, even if there are no actual changes to apply.
```

**Review by mdbirnstiehl — APPROVED** [review]
Looks good! Just a couple of suggestions.

**Comment by eedugon** [comment]
@rogercoll , or @gizas : I'd like you to review this PR and let me know if there are other configuration settings you believe we should include for customers to consider customizing when installing the operator.

Should we add or describe the default resources used by the operator for the collectors?

Also, what about this sentence?

>  The `namespace` of the installation cannot be changed and must be set to `opentelemetry-operator-system` during the helm chart installation.

Is it accurate? I've added it because I tried to customize the namespace directly in the `helm install` command and the installation failed, so I guess we might have the namespace defined somewhere else in the chart.

If we need to support installation in a different namespace we should probably create a separate issue for it.

**Comment by gizas** [comment]
Line 145: ```$ helm upgrade --install --namespace opentelemetry-operator-system opentelemetry-kube-stack open-telemetry/opentelemetry-kube-stack --values ./resources/kubernetes/operator/helm/values.yaml --version 0.3.0```

Please Update the version to --version 0.3.3

File: docs/kubernetes/operator/README.md [hunk 5]
```diff
@@ -181,18 +183,63 @@ For detailed instructions and examples on how to instrument applications in Kube
...
+
+### Configurable parameters
+
+The following table lists common parameters that might be relevant for your use case:
+
+| `values.yaml` parameter          |     Description      |
+|----------------------------------|----------------------|
+| `clusterName`                    | Sets the `k8s.cluster.name` field in all collected data. The cluster name is automatically detected for `EKS/GKE/AKS` clusters, but it might be useful for other environments. When monitoring multiple Kubernetes clusters, ensure that the cluster name is properly set in all your environments.<br><br>Refer to the [resourcedetection processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md#cluster-name) for more details about cluster name detection. |
+| `collectors.cluster.resources`   | Configures CPU and memory requests and limits applied to the `Deployment` EDOT Collector responsible for cluster-level metrics.<br>This setting follows the standard [Kubernetes resources syntax](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits) for specifying requests and limits. |
```
Comment (gizas) [inline docs/kubernetes/operator/README.md:220]:
```suggestion
| `collectors.cluster.resources`   | Configures `CPU` and `memory` `requests` and `limits` applied to the `Deployment` EDOT Collector responsible for cluster-level metrics.<br>This setting follows the standard [Kubernetes resources syntax](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits) for specifying requests and limits. |
```

Comment (gizas) [inline docs/kubernetes/operator/README.md:220]:
```suggestion
| `collectors.cluster.resources`   | Configures `CPU` and `memory` `requests` and `limits` applied to the `Deployment` EDOT Collector responsible for cluster-level metrics.<br>This setting follows the standard [Kubernetes resources syntax](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits) for specifying requests and limits. |
```

**Comment by eedugon** [comment]
thanks @gizas ! I'll apply the suggested changes :)

**Comment by gizas** [comment]
I am little worried with the copatibility matrix that we can forget to update as we upgrade the versions. Can we make sure that we make this update part of DoD somewhere or automation step?

The link Values files are available in the [resources directory](https://github.com/elastic/opentelemetry/blob/7b8dd032cccbaecb6e541d8d48cd7a512acbb65c/resources/kubernetes/operator/helm) points to a custom reference, I would say to link it with main

Also in the matrix can you update the 8.16.0 values.yaml to have a link to https://github.com/elastic/opentelemetry/blob/8.16/resources/kubernetes/operator/helm/values.yaml and the chart version to be 0.3.3 !!!

For additional things to describe, I would say for now we are ok. I will let other to comment on this

For the namespace I did a smoke test and worked for me

<details>
<summary>Commands to install to new namespace</summary>

kubectl create namespace opentelemetry-operator-system2
kubectl create secret generic elastic-secret-otel --namespace opentelemetry-operator-system2 --from-literal=elastic_endpoint='https://f670e0b954504a56819e3172d9d09af4.us-west2.gcp.elastic-cloud.com:443' --from-literal=elastic_api_key='Yk11ajlEJ4bExEYlBNQQ=='
helm repo add open-telemetry 'https://open-telemetry.github.io/opentelemetry-helm-charts' --force-update
helm upgrade --install opentelemetry-kube-stack open-telemetry/opentelemetry-kube-stack \\n  --namespace opentelemetry-operator-system2 \\n  --values './values.yaml' \\n  --version '0.3.3'
</details>

User needs to:
1. Update https://github.com/elastic/opentelemetry/blob/8.16/resources/kubernetes/operator/helm/values.yaml#L721
2. And for instrumentation the url https://github.com/elastic/opentelemetry/blob/8.16/resources/kubernetes/operator/helm/values.yaml#L936
3. And of course the annotations to have correct namespace later on the manual steps of annotating the deployments' apps

But I agree lets create a new issue to test namespace change. And for now lets remove any reference to namespace configuration. WDYT?

**Comment by eedugon** [comment]
@gizas , thanks a lot for your comments!
For all the `values.yaml` related comments I was going to address that in a different PR (as soon as it was clear where exactly the values file is published), but I think I can manage it on this PR too.

Replying inline:

> I am little worried with the compatibility matrix that we can forget to update as we upgrade the versions. Can we make sure that we make this update part of DoD somewhere or automation step?

Good point, let's discuss it in a different issue. I will create one.

> The link Values files are available in the [resources directory](https://github.com/elastic/opentelemetry/blob/7b8dd032cccbaecb6e541d8d48cd7a512acbb65c/resources/kubernetes/operator/helm) points to a custom reference, I would say to link it with main
> Also in the matrix can you update the 8.16.0 values.yaml to have a link to https://github.com/elastic/opentelemetry/blob/8.16/resources/kubernetes/operator/helm/values.yaml and the chart version to be 0.3.3 !!!

Perfect, I'll work on that on this PR as it's a small and needed change.

> For the namespace I did a smoke test and worked for me

Cool, I'm removing then the `NOTE` about not being supported and we can add the procedure to change the namespace in a different issue + PR (if we have a way to not hardcode the namespace in the `values.yaml` that would be ideal :) ).

Thanks a lot for your help!

**Comment by eedugon** [comment]
Changes implemented @gizas , feel free to take a look.

**Review by gizas — APPROVED** [review]

File: docs/kubernetes/operator/README.md [hunk 6]
```diff
@@ -121,27 +123,37 @@ Notes:
...
 
-```
-$ helm repo add open-telemetry https://open-telemetry.github.io/opentelemetry-helm-charts
-$ helm repo update
-$ helm upgrade --install --namespace opentelemetry-operator-system opentelemetry-kube-stack open-telemetry/opentelemetry-kube-stack --values ./resources/kubernetes/operator/helm/values.yaml --version 0.3.0
-```
+    - At this repository main branch ([main values.yaml](https://raw.githubusercontent.com/elastic/opentelemetry/refs/heads/main/resources/kubernetes/operator/helm/values.yaml)). This includes the latest configuration settings but might not be fully tested.
+    - (**recommended**) At different release named branches, for example the [8.16 released values.yaml](https://raw.githubusercontent.com/elastic/opentelemetry/refs/heads/8.16/resources/kubernetes/operator/helm/values.yaml). Check [compatibility matrix](#compatibility-matrix) for a complete list of available manifests.
+
+4. Run the following commands to deploy the Helm Chart, using the appropriate values file.
```
Comment (eedugon) [inline docs/kubernetes/operator/README.md:148]:
```suggestion
4. Run the following commands to deploy the Helm Chart, using the appropriate values file:
```

File: docs/kubernetes/operator/README.md [hunk 7]
```diff
@@ -121,27 +123,37 @@ Notes:
...
+
+4. Run the following commands to deploy the Helm Chart, using the appropriate values file:
+
+    ```bash
+    helm repo add open-telemetry https://open-telemetry.github.io/opentelemetry-helm-charts
+    helm repo update
+    helm upgrade --install --namespace opentelemetry-operator-system opentelemetry-kube-stack open-telemetry/opentelemetry-kube-stack \
+          --values 'https://raw.githubusercontent.com/elastic/opentelemetry/refs/heads/8.16/resources/kubernetes/operator/helm/values.yaml' \
+          --version 0.3.3
+    ```
```
Comment (eedugon) [inline docs/kubernetes/operator/README.md:156]: @mdbirnstiehl , @gizas  : would you please review this new step I have added?
I forgot to mention the possibility of customizing the configuration in the manual installation steps.

**Pull request merged**